In [1]:
import unittest
import numpy as np
import numbers

from Ensembler.src import potentials
from Ensembler.src import system, integrator



In [17]:
#prepare:
integ = integrator.monteCarloIntegrator()
pot = potentials.OneD.harmonicOsc1D()
conditions = []
temperature = 300
position = [0.1]
mass = [1]
steps = 1000

sys = system.system(potential=pot, integrator=integ, position=position, temperature=temperature)

In [14]:
%%timeit
sys.simulate(steps=steps)

26.5 ms ± 763 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)


# History of Performance
## Monte Carlo HarmoicOsc 1D

   * start:
     * Steps 100:  2.74 ms ± 33.4 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
     * Steps 1000: 27.4 ms ± 258 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
     * Steps 10000: 273 ms ± 2.97 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

         

In [8]:
#prepare Lambd
integ = integrator.monteCarloIntegrator()
ha = potentials.OneD.harmonicOsc1D(x_shift=-5)
hb = potentials.OneD.harmonicOsc1D(x_shift=5)
pot = potentials.OneD.linCoupledHosc(ha=ha, hb=hb)

temperature = 300
position = [0]
lam = 0

sys = system.perturbedSystem(potential=pot, integrator=integ, position=position, temperature=temperature, lam=lam)

steps=10000


In [9]:
%%timeit
sys.simulate(steps=steps, initSystem=False, withdrawTraj=True)  #withdrawTraj is needed in the context because of the interaction between different Tests

Progress: 	0%	10%	20%	30%	40%	50%	60%	70%	80%	90%	100%
Progress: 	0%	10%	20%	30%	40%	50%	60%	70%	80%	90%	100%
Progress: 	0%	10%	20%	30%	40%	50%	60%	70%	80%	90%	100%
Progress: 	0%	10%	20%	30%	40%	50%	60%	70%	80%	90%	100%
Progress: 	0%	10%	20%	30%	40%	50%	60%	70%	80%	90%	100%
Progress: 	0%	10%	20%	30%	40%	50%	60%	70%	80%	90%	100%
Progress: 	0%	10%	20%	30%	40%	50%	60%	70%	80%	90%	100%
Progress: 	0%	10%	20%	30%	40%	50%	60%	70%	80%	90%	100%
417 ms ± 12.3 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


# History of Performance
## Monte Carlo Perturbed 2 HarmoicOsc 1D

   * start:
     * Steps 100:  4.24 ms ± 675 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
     * Steps 1000: 37.9 ms ± 374 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
     * Steps 10000: 417 ms ± 12.3 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
         